webapp/store/views.py

```diff
@@ -162,29 +156,71 @@ def search_snap():
         except ApiError as api_error:
             status_code, error_info = _handle_errors(api_error)
 
+        total_pages = None
+
         if "total" in searched_results:
             total_results_count = searched_results["total"]
+            total_pages = ceil(total_results_count / size)
         else:
             total_results_count = None
 
         snaps_results = logic.get_searched_snaps(searched_results)
-        links = logic.get_pages_details(
-            flask.request.base_url,
-            (
-                searched_results["_links"]
-                if "_links" in searched_results
-                else []
-            ),
-        )
+
+        links = {}
+
+        if page > 1:
+            links["first"] = logic.build_pagination_link(
+                snap_searched=snap_searched,
+                snap_category=snap_category,
+                page=1,
+            )
+            links["prev"] = logic.build_pagination_link(
+                snap_searched=snap_searched,
+                snap_category=snap_category,
+                page=page - 1,
+            )
+
+        if not total_pages or page < total_pages:
+            links["next"] = logic.build_pagination_link(
+                snap_searched=snap_searched,
+                snap_category=snap_category,
+                page=page + 1,
+            )
+            if total_pages:
+                links["last"] = logic.build_pagination_link(
+                    snap_searched=snap_searched,
+                    snap_category=snap_category,
+                    page=total_pages,
+                )
+
+        featured_snaps = []
+
+        number_of_featured_snaps = 19
```

Comment:
One more magic number worth explaining in comments (especially that it depends on layout, so when design is updated we need to take that into account).